cpp/velox/operators/reader/ParquetReaderIterator.cc

```diff
@@ -39,11 +39,11 @@ std::shared_ptr<facebook::velox::common::ScanSpec> makeScanSpec(const facebook::
 ParquetReaderIterator::ParquetReaderIterator(
     const std::string& path,
     int64_t batchSize,
-    facebook::velox::memory::MemoryPool* pool)
-    : FileReaderIterator(path), pool_(pool), batchSize_(batchSize) {}
+    std::shared_ptr<facebook::velox::memory::MemoryPool> pool)
+    : FileReaderIterator(path), pool_(std::move(pool)), batchSize_(batchSize) {}
 
 void ParquetReaderIterator::createRowReader() {
-  facebook::velox::dwio::common::ReaderOptions readerOptions{pool_};
+  facebook::velox::dwio::common::ReaderOptions readerOptions{pool_.get()};
```

Comment:
Why change the pool to std::shared_ptr<facebook::velox::memory::MemoryPool> pool

Reply:
https://github.com/apache/incubator-gluten/pull/9483/files#diff-921be51ce29eb20bd569880bbcc0d450ff1529e78cbf4d101bc31bf37c9d0291R126-R127 
The reader now owns a child memory pool. With this change we can track the memory usage separately in case the dumping causes OOM.